packages/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectViewMoreTypeaheadGroupedDemo.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import { Select, SelectOption, SelectVariant, SelectGroup, Checkbox, StackItem, Title } from '@patternfly/react-core';
+
+/* eslint-disable no-console */
+interface TypeAheadOption {
+  value?: string;
+  disabled?: boolean;
+}
+
+/* eslint-disable no-console */
+export interface SelectViewMoreTypeaheadGroupedDemoState {
+  isOpen: boolean;
+  selected: string[];
+  numOptions: number;
+  isLoading: boolean;
+  isCreatable: boolean;
+  newOptions: boolean;
+  inputValuePersisted: boolean;
+  options: TypeAheadOption[];
+}
+
+export class SelectViewMoreTypeaheadGroupedDemo extends React.Component<SelectViewMoreTypeaheadGroupedDemoState> {
+  static displayName = 'SelectViewMoreTypeaheadGroupedDemo';
+  state = {
+    isOpen: false,
+    selected: [] as string[],
+    numOptions: 1,
+    isLoading: false,
+    isCreatable: false,
+    newOptions: false,
+    inputValuePersisted: false
+  };
+
+  options = [
```

Comment:
Per our discussion on slack, `options` should be managed in the state since we have option creation functionality.